internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -206,20 +282,53 @@ func (p *processRunner) run(eventSink events.Recorder) error {
 		defer reaper.Stop(notifyCh)
 	}
 
-	err = cmdWrapper.cmd.Start()
-
-	cmdWrapper.afterStart()
-
+	pid, err := cmdWrapper.launcher.Launch(&cmdWrapper)
 	if err != nil {
 		return fmt.Errorf("error starting process: %w", err)
 	}
 
-	eventSink(events.StateRunning, "Process %s started with PID %d", p, cmdWrapper.cmd.Process.Pid)
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return fmt.Errorf("could not find process: %w", err)
+	}
+
+	if err := sys.AdjustOOMScore(pid, p.opts.OOMScoreAdj); err != nil {
+		return fmt.Errorf("failed to change OOMScoreAdj of process %s to %d", p, p.opts.OOMScoreAdj)
+	}
+
+	cgroupPath := cgroup.Path(p.opts.CgroupPath)
+
+	if cgroups.Mode() == cgroups.Unified {
+		cgv2, err := cgroup2.Load(cgroupPath)
+		if err != nil {
+			return fmt.Errorf("failed to load cgroup %s: %s", cgroupPath, err)
```

Comment:
Always use `%w` for variables that implement or themselves are `error` type in `fmt.Errorf`. [Details](https://pkg.go.dev/fmt#Errorf). You can specify it several times.